sycl/doc/extensions/supported/sycl_ext_oneapi_sub_group_mask.asciidoc

```diff
@@ -100,6 +113,38 @@ The mask is defined such that the least significant bit (LSB) corresponds to
 the work-item with id 0, and the most significant bit (MSB) corresponds to the
 work-item with the id `max_local_range()-1`.
 
+|===
+|Constructor|Description
+
+|`sub_group_mask()`
+|Constructs a group mask with all bits set to 0. Size of a group mask
+corresponds to max local range of the sub-group which work-item belongs to.
+
+|`sub_group_mask(unsigned long long val)`
+|Constructs a group mask with the first `N` bit positions to the
+corresponding bit values in _val_. `N` is a size of a group mask and it
+corresponds to max local range of the sub-group which work-item belongs to. If
+size of a group mask is bigger than the number of bits in the value
+representation of `unsigned long long`, the remaining positions are initialized
+to zero.
+
+|`template <typename T, std::size_t K> sub_group_mask(const sycl::marray<T, K>& &val)`
+|Constructs a group mask with the first `N` bit positions to the
+corresponding bit values in _val_. `T` must be a SYCL `marray` of integral
+types. `N` is a size of a group mask and it corresponds to max local range of
+the sub-group which work-item belongs to. If size of a group mask is bigger than
+number of bits in the value representation of `T`, the remaining positions are
+initialized to zero.
+
+|`sub_group_mask(const sub_group_mask &other) = default`
+|Constructs a group mask as a copy of _other_.
+
+|`sub_group_mask& operator=(const sub_group_mask &other) = default`
+|Assigns this instance of `sub_group_mask` with a copy of _other_. Size of both
+group masks must be the same or otherwise behavior is undefined.
```

Comment:
The user has no control over the size of a group mask because the size depends on the number of work-items in the sub-group, and this is implementation-defined.  Should we instead say that `other` must be created for the same sub-group as `this`?

Reply:
User can control sub-group size through `reqd_sub_group_size` attribute or through corresponding property. Also, masks are tied to _maximum_ sub-group size, meaning that size of a mask will be the same for all sub-groups in a kernel, i.e. we can "share" masks between work-items of a kernel just fine.

I'm fine with either variant, but I slightly prefer the existing wording: it describes the actual (kind of) limitation we have and thus provides users with freedom of doing various stuff. At the same time, I don't quite imagine a use case where `sub_group_mask` would be shared/transferred between work-items and therefore perhaps it is ok to "lock" it to a single sub-group.

@Pennycook, any thoughts here?

Reply:
> User can control sub-group size through reqd_sub_group_size attribute or through corresponding property. Also, masks are tied to maximum sub-group size, meaning that size of a mask will be the same for all sub-groups in a kernel

Good point.  I was thinking that the size of the group mask was the number of work-items in the sub-group, but this is not the case.  I see now that it is the *maximum* number of work-items in a sub-group, and this is controllable by the user via `[[reqd_sub_group_size]]`.  I agree that the current wording is fine.

Reply:
> @Pennycook, any thoughts here?

The only copying of sub-group masks I think we need to forbid is between kernels.  That covers the cases where two kernels have different `[[reqd_sub_group_size]]` attributes, and where two kernels are submitted to different devices.

I also agree that the current wording is fine.